sw/device/silicon_creator/manuf/lib/individualize_sw_cfg_functest.c

```diff
@@ -101,6 +102,69 @@ static status_t check_otp_ast_cfg(void) {
   return OK_STATUS();
 }
 
+/**
+ * Check the *SW_CFG partition digests.
+ */
+static status_t check_otp_sw_cfg_digest(dif_otp_ctrl_partition_t partition) {
+  uint64_t expected_digest, actual_digest = 0;
+
+  // Get actual_digest.
+  CHECK_DIF_OK(dif_otp_ctrl_get_digest(&otp_ctrl, partition, &actual_digest));
+
+  // Compute expected_digest.
+  hmac_sha256_init();
+  switch (partition) {
+    case kDifOtpCtrlPartitionCreatorSwCfg: {
+      hmac_sha256_update(
+          (unsigned char *)(TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR +
+                            OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET +
```

Comment:
Should we pre-calculate this base just once instead of re-creating it in each case?
```c
const char * const  base = TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR + OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET;
```

Reply:
fixed